### ISSUE TITLE
Indentation after =

### ISSUE THREAD

**Issue by lucaswerkmeister** [ceylon/ceylon.formatter#37]
Currently, the formatter turns

``` ceylon
value v =
    something;
```

into

``` ceylon
value v =
something;
```

which is obviously wrong. However, if I simply give the `=` an `indentAfter` of 1, that will stack everywhere, which is wrong as well.
Potential solutions:
#### 1. disallow line breaks after `=`

Cheap solution, and I suppose I could live with it. The correct way to format the above would then be

``` ceylon
value v
    = something;
```
#### 2. `indentAfter = 1`, save `context` in a global variable and close it with next token

Ugly because I have to remember to close that context in a thousand places, and dangerous if I forget it in one place. Still, simple solution that requires no changes to the architecture.
#### 3. add notion of “volatile” indentation to FormattingWriter

Basically the same thing as 2, except built into the FormattingWriter: a context marked `volatile = true` isn’t returned, but instead automatically closed on next `writeToken`. Problem: I no longer have the guarantee that there’s no other context to be closed, so `Token`s probably have a set of contexts they close instead of one.
#### 4. completely decouple context stacking and direction of indentation

Currently, `indentBefore` never stacks, `indentAfter` always does. I could add two flags `stackIndentBefore`, -`After`, and make `false` and `true` the defaults. `=` would then have `stackIndentAfter = false`. 3 is basically one half of this.

I’m leaning towards solution 3.
